File: leetcode/leetcode_p201/code_201.cpp
// 力扣201题，数字范围按位与
#include <iostream>

int main()
{
    using namespace std;
    int m = 7;
    int n = 5;
    int shift = 0;

    while (m != n)
    {
        m = m >> 1;
        n = n >> 1;
        shift ++;
    }
    int result = m << shift;
    cout << "result = " << result << endl;
    return 0;
}

File: leetcode/leetcode_p201/code_201.py
#  力扣201题，数字范围按位与
m = 7
n = 5

shift = 0
while m != n:
    m = m >> 1
    n = n >> 1
    shift += 1

print(m, n, shift)
m = m << shift
print("result:", m)